Sources/TuistKit/Services/GraphService.swift

```diff
@@ -1,36 +1,60 @@
+import DOT
 import Foundation
+import GraphViz
 import TSCBasic
 import TuistGenerator
 import TuistLoader
 import TuistSupport
 
 final class GraphService {
     /// Dot graph generator.
-    private let dotGraphGenerator: DotGraphGenerating
+    private let graphVizGenerator: GraphVizGenerating
 
     /// Manifest loader.
     private let manifestLoader: ManifestLoading
 
-    init(dotGraphGenerator: DotGraphGenerating = DotGraphGenerator(modelLoader: GeneratorModelLoader(manifestLoader: ManifestLoader(),
+    init(graphVizGenerator: GraphVizGenerating = GraphVizGenerator(modelLoader: GeneratorModelLoader(manifestLoader: ManifestLoader(),
                                                                                                      manifestLinter: ManifestLinter())),
          manifestLoader: ManifestLoading = ManifestLoader()) {
-        self.dotGraphGenerator = dotGraphGenerator
+        self.graphVizGenerator = graphVizGenerator
         self.manifestLoader = manifestLoader
     }
 
-    func run(skipTestTargets: Bool, skipExternalDependencies: Bool) throws {
-        let graph = try dotGraphGenerator.generate(at: FileHandler.shared.currentPath,
-                                                   manifestLoader: manifestLoader,
-                                                   skipTestTargets: skipTestTargets,
-                                                   skipExternalDependencies: skipExternalDependencies)
+    func run(format: GraphFormat,
+             skipTestTargets: Bool,
+             skipExternalDependencies: Bool,
+             path: String?) throws {
+        let graphVizGraph = try graphVizGenerator.generate(at: FileHandler.shared.currentPath,
+                                                           manifestLoader: manifestLoader,
+                                                           skipTestTargets: skipTestTargets,
+                                                           skipExternalDependencies: skipExternalDependencies)
+        let basePath = path != nil ? path! : FileHandler.shared.currentPath.pathString
```

Comment:
To get the correct `path` we'd need something like what's used in `GenerateService` which caters for both absolute paths as well as relative paths.

e.g.

```swift
private func path(_ path: String?) -> AbsolutePath {
        if let path = path {
            return AbsolutePath(path, relativeTo: FileHandler.shared.currentPath)
        } else {
            return FileHandler.shared.currentPath
        }
    }
```

Reply:
👍